cmd/zdb/zdb.c

```diff
@@ -518,7 +523,8 @@ dump_bpobj(objset_t *os, uint64_t object, void *data, size_t size)
 			(void) printf("got error %u from dmu_read\n", err);
 			break;
 		}
-		snprintf_blkptr_compact(blkbuf, sizeof (blkbuf), &bp);
+		boolean_t free = BP_GET_FREE(&bp);
+		snprintf_blkptr_compact(blkbuf, sizeof (blkbuf), &bp, free);
```

Comment:
There doesn't seem to be much point to having a separate variable there -- it's not used anywhere else, and it uses the name of a standard C function. I'd suggest just changing the function call to have the BP_GET_FREE(&bp) call inline.